api/v1alpha1/envoyproxy_types.go

```diff
@@ -342,6 +342,12 @@ type EnvoyProxyProvider struct {
 	//
 	// +optional
 	Kubernetes *EnvoyProxyKubernetesProvider `json:"kubernetes,omitempty"`
+	// Custom provides configuration for running the data plane, e.g. Envoy proxy.
+	// If unspecified and type is "Custom", default settings for the custom provider
+	// are applied.
+	//
+	// +optional
+	Custom *EnvoyProxyCustomProvider `json:"custom,omitempty"`
```

Comment:
imo this should be `Host`

the `Custom` provider was added in the `EnvoyGateway` API to seperate out Resource Provider and Infrastructure provider (`Host` is a infrastructure provider) and the `EnvoyProxy` API is the API to manage infrastructure proxy fleet settings, so `Host` makes more sense here

wdyt @envoyproxy/gateway-maintainers ?

Reply:
To me, `Host` holds semantics about the "host environment" (where the Envoy process would run), but not about the Envoy process itself. This is why I opted to use the existing Custom option.
I'm open to a better name, though, as Custom doesn't carry any semantics (I just feel Host is not the best choice here?)

Reply:
Had an offline discussion with @arkodg to get a better understanding of the current semantics for Host, and I have more clarity now and `Host` feels right for EP as well.